### PR TITLE
Migrate Assembly API

### PR DESCRIPTION
**Pull request by kamikazebr**
- [x] Migrated AssemblyAPI
- [x] Changes XDAI to Gnosis
- [x] Some config files to build graph-node locally and hardhat forks

### Refs
https://thegraph.com/docs/en/release-notes/assemblyscript-migration-guide/

### CODE REVIEW

File: hardhat/hardhat.config.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
+/**
+ * @type import('hardhat/config').HardhatUserConfig
+ */
+module.exports = {
+  solidity: '0.7.3',
+  networks: {
+    hardhat: {
+      forking: {
+        // url: "https://xdai-archive.blockscout.com",
```
Comment (Corantin) [inline hardhat/hardhat.config.js]: comented code

File: package.json [hunk 2]
```diff
@@ -15,13 +15,13 @@
     "deploy:xdai": "graph deploy 1hive/honeyswap-xdai subgraph.xdai.yaml --ipfs https://api.thegraph.com/ipfs/ --node https://api.thegraph.com/deploy/ --debug"
   },
   "devDependencies": {
-    "@graphprotocol/graph-cli": "^0.19.0",
-    "@graphprotocol/graph-ts": "^0.19.0",
+    "@graphprotocol/graph-cli": "0.37.2",
+    "@graphprotocol/graph-ts": "^0.29.1",
```
Comment (Corantin) [inline package.json]: Could be worh it to get latest here with is "@graphprotocol/graph-ts@^0.29.3"
```suggestion
    "@graphprotocol/graph-ts": "^0.29.3",
```

File: src/mappings/core.ts [hunk 3]
```diff
@@ -41,6 +42,9 @@ export function handleTransfer(event: Transfer): void {
   }
 
   let factory = HoneyswapFactory.load(getFactoryAddress())
+  if (!factory) {
+    return
```
Comment (Corantin) [inline src/mappings/core.ts:50]: maybe need to log warning in that case

File: src/mappings/core.ts [hunk 4]
```diff
@@ -51,6 +55,9 @@ export function handleTransfer(event: Transfer): void {
 
   // get pair and load contract
   let pair = Pair.load(event.address.toHexString())
+  if (!pair) {
+    return
```
Comment (Corantin) [inline src/mappings/core.ts:65]: same ?

Comment (Corantin) [reply src/mappings/core.ts:65]: NOt sure if we want to silent these edge case that could result in incomprehension state

File: src/mappings/core.ts [hunk 5]
```diff
@@ -287,15 +305,25 @@ export function handleSync(event: Sync): void {
 
 export function handleMint(event: Mint): void {
   let transaction = Transaction.load(event.transaction.hash.toHexString())
+  if (!transaction) {
+    return
+  }
   let mints = transaction.mints
```
Comment (Corantin) [inline src/mappings/core.ts:322]: If previously we assumed that is not null, we can either use the ! operator or log a warning that the object is null

**Review by Corantin — CHANGES_REQUESTED** [review]

File: src/mappings/core.ts (hunk 3 above)
Comment (kamikazebr) [reply src/mappings/core.ts:50]: Fixed

File: src/mappings/core.ts (hunk 4 above)
Comment (kamikazebr) [reply src/mappings/core.ts:65]: Fixed

File: src/mappings/core.ts (hunk 5 above)
Comment (kamikazebr) [reply src/mappings/core.ts:322]: Fixed

**Comment by kamikazebr** [comment]
Did all changes, but looks like Github still dont allow to merge. Something wrong with the requested changes, they already outdated.

**Pull request merged**